tests/test_athena.py

```diff
@@ -282,24 +282,51 @@ def test_athena_ctas_empty(glue_database):
     assert len(list(wr.athena.read_sql_query(sql=sql, database=glue_database, chunksize=1))) == 1
 
 
-@pytest.mark.xfail()
-def test_athena_struct(glue_database):
+def test_athena_struct_simple(path, glue_database):
     sql = "SELECT CAST(ROW(1, 'foo') AS ROW(id BIGINT, value VARCHAR)) AS col0"
     with pytest.raises(wr.exceptions.UnsupportedType):
         wr.athena.read_sql_query(sql=sql, database=glue_database, ctas_approach=False)
+    # CTAS approach
     df = wr.athena.read_sql_query(sql=sql, database=glue_database, ctas_approach=True)
     assert len(df.index) == 1
     assert len(df.columns) == 1
     assert df["col0"].iloc[0]["id"] == 1
     assert df["col0"].iloc[0]["value"] == "foo"
-    sql = "SELECT ROW(1, ROW(2, ROW(3, '4'))) AS col0"
+    # UNLOAD approach
+    df = wr.athena.read_sql_query(
+        sql=sql, database=glue_database, ctas_approach=False, unload_approach=True, s3_output=path
+    )
+    assert len(df.index) == 1
+    assert len(df.columns) == 1
+    assert df["col0"].iloc[0]["id"] == 1
+    assert df["col0"].iloc[0]["value"] == "foo"
```

Comment:
would asserting that the above df with ctas is equal to the one with unload instead of rewriting the same assertions a second time?

Reply:
No - what if both are broken? 🙂

Reply:
Sorry I was not clear. I meant keep the first assert statements as is for the first df (ctas)
but assert that the two dataframes are equal for the second one. If both are broken then the first assert statements will fail.

Reply:
Done